### PR TITLE
📖 doc: extend quickstart to support ClusterClass

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
This PR extends our current quickstart documentation to also cover clusters with ClusterClass

Note: This PR intentionally only extends the existing quickstart, we now have 2 quickstarts supporting ClusterClass, but this will be cleaned up in later PRs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -172,6 +172,17 @@ and `kubeadm` control-plane providers.
...
 #### Initialization for common providers
 
+<aside class="note">
+
+<h1>ClusterClass</h1>
+
+If you want to deploy a Cluster with ClusterClass, the `ClusterTopology` feature has to be enabled.
+```bash
+export CLUSTER_TOPOLOGY=true
+```
```
Comment (fabriziopandini) [inline docs/book/src/user/quick-start.md:182]: Let's make this more generic by having a sub paragraph named "enabling feature gated features" that links to the feature gate doc, but keeping ClusterClass as example

Comment (sbueringer) [reply docs/book/src/user/quick-start.md:180]: Done, ptal

File: docs/book/src/user/quick-start.md [hunk 2]
```diff
@@ -170,6 +170,16 @@ The command accepts as input a list of providers to install; when executed for t
 automatically adds to the list the `cluster-api` core provider, and if unspecified, it also adds the `kubeadm` bootstrap
 and `kubeadm` control-plane providers.
 
+#### Enabling Feature Gates
+
+Feature gates can be enabled by exporting environment variables before executing `clusterctl init`. 
+For example, the `ClusterTopology` feature, which is required to managed Clusters with ClusterClass,
+can be enabled via:
```
Comment (fabriziopandini) [inline docs/book/src/user/quick-start.md:177]: nit
```suggestion
For example, the `ClusterTopology` feature, which is required to enable support for managed topologies and ClusterClass,
can be enabled via:
```

Comment (sbueringer) [reply docs/book/src/user/quick-start.md]: Done

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**